### PR TITLE
TS: allow namespace imports as types

### PR DESCRIPTION
**Pull request by asger-semmle**
Apparently I was very careful to *not* allow namespaces to be seen as types, but apparently now they can. Fixes some FPs reported by our unused import query.

### CODE REVIEW

File: javascript/extractor/tests/esnext/output/trap/yield-import.js.trap [hunk 1]
```diff
@@ -9,169 +9,35 @@ hasLocation(#10000,#10002)
...
 scopes(#20000,0)
 #20001=@"script;{#10000},1,1"
-#20002=*
-lines(#20002,#20001,"function* f() {","
+toplevels(#20001,0)
+#20002=@"loc,{#10000},1,1,1,1"
+locations_default(#20002,#10000,1,1,1,1)
+hasLocation(#20001,#20002)
+#20003=*
+jsParseErrors(#20003,#20001,"Error: Unexpected token","  yield import(""foo"")
```
Comment (xiemaisi) [inline javascript/extractor/tests/esnext/output/trap/yield-import.js.trap]: This doesn't look right; that syntax error shouldn't be there on the rc.

**Comment by esben-semmle** [comment]
Can you check if it solves this https://jira.semmle.com/browse/ODASA-7792 issue as well?

File: javascript/extractor/tests/esnext/output/trap/yield-import.js.trap (hunk 1 above)
Comment (asger-semmle) [reply javascript/extractor/tests/esnext/output/trap/yield-import.js.trap]: *Sigh*. That was a case of Eclipse needing a refresh

**Comment by asger-semmle** [comment]
> Can you check if it solves this https://jira.semmle.com/browse/ODASA-7792 issue as well?

That was fixed previosly. I've closed the issue.

**Pull request merged**